circuit-foundation/src/jvmTest/kotlin/com/slack/circuit/foundation/KeyedCircuitContentTests.kt

```diff
@@ -8,6 +8,7 @@ import androidx.compose.runtime.Composable
 import androidx.compose.runtime.MutableState
```

Comment:
testTwoB relies on the fact that `rememberRetained` doesn't retained/store on each screen change, as it is not ran in a `NavigableCircuitContent`. Retained state isn't the point of this test, so migrated it to use `remember` instead which is the point.